### ISSUE TITLE
soft update runs every gradient step regardless of policy_delay

### ISSUE THREAD

**Issue by httpsVishu** [araffin/sbx#87]
hey, was going through the `_train` loop in `sbx/sac/sac.py` file and noticed something that looks off
the soft update of the target network runs unconditionally every gradient step
```python
qf_state = cls.soft_update(tau, qf_state)  # always runs
```
but the actor update is correctly gated behind `policy_delay`
```python
actor_state, qf_state, ... = jax.lax.cond(
    (policy_delay_offset + i) % policy_delay == 0,
    cls.update_actor_and_temperature,
    ...
)
```
so when `policy_delay > 1` (like in the droq config), the target network ends up being updated `policy_delay` times more frequently than the actor, that goes against the intended stabilization from delayed updates like the whole point of `policy_delay` is to keep the target stable relative to the actor not just delay the actor alone.

should not the soft update be inside the same `cond` as the actor update or at least gated similarly? happy to put up a pr if this looks like a real issue

**Comment by araffin** [comment]
if you read the droq/redq papers and their implementations, the current sbx implementation is correct.
